lib/features/steps/request_assertion_steps.rb

```diff
@@ -226,4 +226,28 @@ def read_key_path hash, key_path
     assert_not_nil(read_key_path(element, element_key_path),
            "Each element in '#{key_path}' must have '#{element_key_path}'")
   end
+end
+
+# Checks a given payload field for the current request is the same as a payload field in the next request
+#
+# @param current_field [String] the payload element from the current request
+# @param next_field [String] the payload element in the next request
+Then("the payload field {string} in the current request matches the payload field {string} in the next request") do |current_field, next_field|
```

Comment:
could another approach be instead to read a payload field into a global variable or something? We could have a step like `'the payload field is stored in {string}'` which stores it in `@payload_fields[string] = value` and then in subsequent steps we can check fields are equal to that? Just seems a little more reusable to do something like that than to assume its the next payload.

Reply:
We could do, I was reluctant to add that sort of complexity.  I guess so long as it's abstracted out enough it should be useable without being clunky.